src/base/reports/ConsoleSink.hpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Signal Estimator authors
+// Licensed under MIT
+
+#pragma once
+
+#include <spdlog/fmt/bundled/format.h>
+
+namespace signal_estimator {
+
+class ConsoleSink final {
```

Comment:
Since in audio domain sink (and source) terms are used frequently for a different thing (audio devices & files), using ISink/ConsoleSink here can be a bit confusing. I suggest to select some different term for this.

Reply:
Ok, we can think about different naming.

I opted for the term sink because that is the terminology which is also used by spdlog for its sinks.

Reply:
Yep, I also immediately though about spdlog sinks :)

Reply:
So we have two conflicting problem domains here. From the point of audio, the term sink might be confusing, but from the point of a logging library such as spdlog the term sink is appropriate for the current use.

What solution would you prefer?

Reply:
Maybe something like:

* ITextSink
* ITextOutput

?

Reply:
The sink interface is used by both reporters, `TextReporter` and `JsonReporter`. I think that if you mangle the term `Text` into the interface name it could be confusing as well.

Reply:
I see. So it's actually used by TextPrinter and JsonPrinter, right? Then maybe IPrintOutput or IPrintSink, or something like this?